adoc/extensions/sycl_khr_free_function_commands.adoc

```diff
@@ -0,0 +1,786 @@
+void fill(sycl::handler& h, T* ptr, const T& pattern, size_t count); (1)
+
+template <typename T, int Dims, access_mode Mode>
+void fill(sycl::handler& h,
+          sycl::accessor<T, Dims, Mode, target::device> dest,
+          const T& src);                                             (2)
+
+template <typename T>
+void fill(sycl::queue q, T* ptr, const T& pattern, size_t count);    (3)
+
+template <typename T, int Dims, access_mode Mode>
+void fill(sycl::queue q,
+          sycl::accessor<T, Dims, Mode, target::device> dest,
+          const T& src);                                             (4)
+
+}
+----
+
+_Constraints (1, 3)_: [code]#T# is <<device-copyable>>.
+
+_Effects (1)_: Equivalent to [code]#h.fill(ptr, pattern, count)#.
+
+_Effects (2)_: Equivalent to [code]#h.fill(dest, src)#.
+
+_Effects (3)_: Equivalent to [code]#q.submit([&](handler& h) { fill(h, ptr,
+pattern, count); })#.
+
+_Effects (4)_: Equivalent to [code]#q.submit([&](handler& h) { fill(h, dest,
+src); })#.
+
+'''
+
+.[apititle]#update_host#
+[source,role=synopsis,id=api:update_host]
+----
+namespace sycl::khr {
+
+template <typename T, int Dims, access_mode Mode>
+void update_host(sycl::handler& h, accessor<T, Dims, Mode, target::device> acc); (1)
+
+template <typename T, int Dims, access_mode Mode>
+void update_host(sycl::queue q, accessor<T, Dims, Mode, target::device> acc);    (2)
+
+}
+----
+_Effects (1)_: Equivalent to [code]#h.update_host(acc)#.
+
+_Effects (2)_: Equivalent to [code]#q.submit([&](handler& h) { update_host(h,
+acc); })#.
+
+'''
+
+.[apititle]#prefetch#
+[source,role=synopsis,id=api:prefetch]
+----
+namespace sycl::khr {
+
+void prefetch(sycl::handler& h, void* ptr, size_t numBytes); (1)
+
+void prefetch(sycl::queue q, void* ptr, size_t numBytes);    (2)
+
+}
+----
+_Effects (1)_: Equivalent to [code]#h.prefetch(ptr, numBytes)#.
+
+_Effects (2)_: Equivalent to [code]#q.submit([&](handler& h) { prefetch(h, ptr,
+numBytes); })#.
+
+'''
+
+.[apititle]#mem_advise#
+[source,role=synopsis,id=api:mem_advise]
+----
+namespace sycl::khr {
+
+void mem_advise(sycl::handler& h, void* ptr, size_t numBytes, int advice); (1)
+
+void mem_advise(sycl::queue q, void* ptr, size_t numBytes, int advice);    (2)
+
+}
+----
+_Effects (1)_: Equivalent to [code]#h.mem_advise(ptr, numBytes, advice)#.
+
+_Effects (2)_: Equivalent to [code]#q.submit([&](handler& h) { mem_advise(h,
+ptr, numBytes, advice); })#.
+
+'''
+
+=== Command and event barriers
+
+.[apititle]#command_barrier#
+[source,role=synopsis,id=api:command_barrier]
+----
+namespace sycl::khr {
+
+void command_barrier(sycl::handler& h); (1)
+
+void command_barrier(sycl::queue q);    (2)
+
+}
+----
+_Effects_: Enqueues a command barrier.
+Any commands submitted after this barrier cannot begin execution until:
+
+* All commands previously submitted to this queue have completed; and
+* All commands associated with this command's dependencies (e.g., via
+  [code]#handler::depends_on#) have completed.
+
```

Comment:
Since `event_barrier` has a clarification for "no-op" case, shall we add one here too?

{note}Overload (1) is no-op when submitted to an in-order queue with no other dependencies (e.g., specified by [code]#handler::depends_on#). Overload (2) is no-op if [code]#queue# is in-order.{endnote}

Reply:
I've added a similar note in 165d07e13b7cfe379c44fe4da4544ed60ab05e69, but weakened from "is" to "may be" (for both notes) in be56bb77ae628ffafbcbd783c251b32aef51e70e.

Even though this is non-normative, upon reflection it seems a bad idea to guarantee that either of these things *is* a no-op, since that implies calling these functions does absolutely nothing and/or has no overhead. That might be true of some implementations, some of the time, but there will probably still be corner cases where _something_ has to happen (e.g., because the user requested an `event`, because the implementation missed an optimization opportunity, etc).